shell/platform/windows/external_texture_gl.cc

```diff
@@ -0,0 +1,134 @@
+typedef void (*glDeleteTexturesProc)(GLsizei n, const GLuint* textures);
+typedef void (*glBindTextureProc)(GLenum target, GLuint texture);
+typedef void (*glTexParameteriProc)(GLenum target, GLenum pname, GLint param);
+typedef void (*glTexImage2DProc)(GLenum target,
+                                 GLint level,
+                                 GLint internalformat,
+                                 GLsizei width,
+                                 GLsizei height,
+                                 GLint border,
+                                 GLenum format,
+                                 GLenum type,
+                                 const void* data);
+
+// A struct containing pointers to resolved gl* functions.
+struct GlProcs {
+  glGenTexturesProc glGenTextures;
+  glDeleteTexturesProc glDeleteTextures;
+  glBindTextureProc glBindTexture;
+  glTexParameteriProc glTexParameteri;
+  glTexImage2DProc glTexImage2D;
+  bool valid;
+};
+
+static const GlProcs& GlProcs() {
+  auto glGenTextures =
+      reinterpret_cast<glGenTexturesProc>(eglGetProcAddress("glGenTextures"));
+  auto glDeleteTextures = reinterpret_cast<glDeleteTexturesProc>(
+      eglGetProcAddress("glDeleteTextures"));
+  auto glBindTexture =
+      reinterpret_cast<glBindTextureProc>(eglGetProcAddress("glBindTexture"));
+  auto glTexParameteri = reinterpret_cast<glTexParameteriProc>(
+      eglGetProcAddress("glTexParameteri"));
+  auto glTexImage2D =
+      reinterpret_cast<glTexImage2DProc>(eglGetProcAddress("glTexImage2D"));
+
+  auto valid = glBindTexture && glDeleteTextures && glBindTexture &&
+               glTexParameteri && glTexImage2D;
+
+  const static struct GlProcs procs = {
+      glGenTextures,   glDeleteTextures, glBindTexture,
+      glTexParameteri, glTexImage2D,     valid,
+  };
+
+  return procs;
+}
+
+}  // namespace
+
+namespace flutter {
+
+struct ExternalTextureGLState {
+  GLuint gl_texture = 0;
+};
+
+ExternalTextureGL::ExternalTextureGL(
+    FlutterDesktopTextureCallback texture_callback,
+    void* user_data)
+    : state_(std::make_unique<ExternalTextureGLState>()),
+      texture_callback_(texture_callback),
+      user_data_(user_data) {}
+
+ExternalTextureGL::~ExternalTextureGL() {
+  const auto& gl = GlProcs();
+  if (gl.valid && state_->gl_texture != 0) {
+    gl.glDeleteTextures(1, &state_->gl_texture);
+  }
+}
+
+bool ExternalTextureGL::PopulateTexture(size_t width,
+                                        size_t height,
+                                        FlutterOpenGLTexture* opengl_texture) {
+  if (!CopyPixelBuffer(width, height)) {
+    std::cerr << "Failed to copy pixel buffer from provider." << std::endl;
+    return false;
+  }
+
+  // Populate the texture object used by the engine.
+  opengl_texture->target = GL_TEXTURE_2D;
+  opengl_texture->name = state_->gl_texture;
+  opengl_texture->format = GL_RGBA8;
+  opengl_texture->destruction_callback = (VoidCallback) nullptr;
```

Comment:
C-style casts are forbidden by the C++ style guide; this should use the appropriate C++ cast. (Does it actually need a cast? IIRC `nullptr` can be assigned to a function pointer directly).